### PR TITLE
NETBEANS-84: Support for JSF 2.3; CDI Changes for JSF Artifact Injection

### PR DESCRIPTION
**Pull request by juneau001**
- Updated files to add support for JSF 2.3.9
- Provides changes to web.beans to enable the injection of Faces resources

### CODE REVIEW

**Comment by geertjanw** [comment]
Can you provide some screenshots here of what can be done as a result of this PR? 

From my point of view, this looks good and thanks for doing this, much needed.

**Comment by juneau001** [comment]
Here are a couple of screenshots showing the provided functionality.

Support for JSF 2.3 tags within editor:
jsf-functionality1

Injection of resources:
Screen Shot 2019-11-22 at 6 06 18 AM

**Comment by ggam** [comment]
Thanks a lot @juneau001

**Comment by lkishalmi** [comment]
For me this looks sane. Any reason not to merge it?

**Comment by ggam** [comment]
Will this go into NB 11.3?

File: enterprise/web.jsf20/nbproject/project.properties [hunk 1]
```diff
@@ -17,6 +17,6 @@
 
 javac.compilerargs=-Xlint -Xlint:-serial
 javac.source=1.7
-release.external/javax.faces-2.2.7.jar=modules/ext/jsf-2_2/javax.faces.jar
-release.external/javax.faces-2.2.7-license.txt=modules/ext/jsf-2_2/license.txt
-spec.version.base=1.39.0
+release.external/javax.faces-2.3.9.jar=modules/ext/jsf-2_2/javax.faces.jar
+release.external/javax.faces-2.3.9-license.txt=modules/ext/jsf-2_2/license.txt
+spec.version.base=1.38.0
```
Comment (matthiasblaesing) [inline enterprise/web.jsf20/nbproject/project.properties]: Spec version needs to be incremented, not decremented (updates are determined by this)

Comment (juneau001) [reply enterprise/web.jsf20/nbproject/project.properties]: Thanks and I apologize for the late response and oversight...I've increased the spec version.

**Comment by jasonex7** [comment]
> We are late in the cycle (i.e. release preparation is already happening). I'd like to see this going in early in the 12.0 cycle.

Is there something that can be done considering this issue has high priority on Jira?

**Comment by ebarboni** [comment]
hi, I guess the code change is done? I will accept if @matthiasblaesing  is ok. Seems ok for me.

**Comment by matthiasblaesing** [comment]
Ok - with the approval of @ebarboni I'm merging this to master, from where it will migrate to the release branch. Thank you.

**Pull request merged**